### PR TITLE
WT-3587 Remove HAVE_VERBOSE conditional compilation

### PR DESCRIPTION
**Pull request by bvpvamsikrishna**
Not ready for review yet.
Need to check the performance implications.

### CODE REVIEW

**Comment by michaelcahill** [comment]
@bvpvamsikrishna do you have performance results for this branch?

**Comment by bvpvamsikrishna** [comment]
@sulabhM @nehakhatri5 request to review this pull request.

File: src/txn/txn_timestamp.c [hunk 1]
```diff
@@ -65,25 +65,20 @@ __wt_timestamp_to_hex_string(
...
  */
 void
 __wt_verbose_timestamp(WT_SESSION_IMPL *session,
     const wt_timestamp_t *ts, const char *msg)
 {
-#ifdef HAVE_VERBOSE
 	char timestamp_buf[2 * WT_TIMESTAMP_SIZE + 1];
 
-	if (__wt_timestamp_to_hex_string(session, timestamp_buf, ts) != 0)
+	if (!WT_VERBOSE_ISSET(session, WT_VERB_TIMESTAMP) ||
```
Comment (sulabhM) [inline src/txn/txn_timestamp.c]: this check is already done by __wt_verbose() called next. We will effectively be doing this test twice now.

Comment (bvpvamsikrishna) [reply src/txn/txn_timestamp.c]: The reason why I added this is reduce performance impact by avoiding the execution of "if" statement before the __wt_verbose call.

**Comment by bvpvamsikrishna** [comment]
@sulabhM Request to check the updates done.

File: src/evict/evict_lru.c [hunk 2]
```diff
@@ -421,13 +417,14 @@ __evict_server(WT_SESSION_IMPL *session, bool *did_work)
 
 	/* Eviction is stuck, check if we have made progress. */
 	if (*did_work) {
-#if defined(HAVE_DIAGNOSTIC) || defined(HAVE_VERBOSE)
-		__wt_epoch(session, &cache->stuck_time);
+#if !defined(HAVE_DIAGNOSTIC)
+		/* Need verbose check only if not in diagnostic build */
+		if (WT_VERBOSE_ISSET(session, WT_VERB_EVICT_STUCK))
 #endif
+			__wt_epoch(session, &cache->stuck_time);
```
Comment (nehakhatri5) [inline src/evict/evict_lru.c]: The function __evict_server gets called repeatedly for eviction pass. So having the __wt_epoch outside of diagnostic mode or in non-verbose mode might have performance impact on eviction thread. Has the eviction thread's performance been compared before and after this change?

**Comment by bvpvamsikrishna** [comment]
@nehakhatri5 Request to check the updates.

File: src/txn/txn_timestamp.c (hunk 1 above)
Comment (sulabhM) [reply src/txn/txn_timestamp.c]: I don't have a good solution to this issue, I think it should be okay to check twice if WT_VERB_TIMESTAMP is set.
@agorrod can you have a look at this change here, what do you think.

**Comment by bvpvamsikrishna** [comment]
@agorrod Request to review this pull request.

File: src/evict/evict_lru.c (hunk 2 above)
Comment (agorrod) [reply src/evict/evict_lru.c]: @nehakhatri5 We ran the Jenkins performance tests, which include a number that stress eviction and there was no noticeable change, so I think this one is safe.

File: src/txn/txn_timestamp.c (hunk 1 above)
Comment (agorrod) [reply src/txn/txn_timestamp.c]: I don't have a strong opinion on this. I think the code is more obvious with the explicit check, so I'm happy for it to stay the way it is here.

**Pull request merged**